### PR TITLE
Adds capability to add namespace in Custom Functions. 

### PR DESCRIPTION
**Pull request by jas-valgotar**
Fixes https://github.com/microsoft/Power-Fx/issues/2002

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs [hunk 1]
```diff
@@ -916,7 +916,7 @@ private class TestFunctionMultiply : CustomTexlFunction
             public override bool IsSelfContained => true;
 
             public TestFunctionMultiply()
-                : base("Func", FunctionCategories.MathAndStat, DType.Number, null, DType.Number, DType.String)
+                : base(DPath.Root, "Func", FunctionCategories.MathAndStat, DType.Number, null, DType.Number, DType.String)
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs:919]: >DPath.Root,  [](http://example.com/codeflow?start=23&length=12)

curious - is this change needed? Can it still call the existing ctor?

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.
